src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_completers.py

```diff
@@ -5,18 +5,20 @@
 
 from azure.cli.core.commands.parameters import get_one_of_subscription_locations
 from azure.cli.core.decorators import Completer
+from azure.mgmt.containerservice.models import ContainerServiceVMSizeTypes
 
 from ._client_factory import cf_compute_service
 
 
 @Completer
 def get_vm_size_completion_list(cmd, prefix, namespace, **kwargs):  # pylint: disable=unused-argument
+    """Return the intersection of the VM sizes allowed by the ACS SDK with those returned by the Compute Service."""
     try:
         location = namespace.location
     except AttributeError:
         location = get_one_of_subscription_locations(cmd.cli_ctx)
     result = get_vm_sizes(cmd.cli_ctx, location)
-    return [r.name for r in result]
+    return sorted(list(set(r.name for r in result) & set(c.value for c in ContainerServiceVMSizeTypes)))
```

Comment:
If ContainerServiceVMSizeTypes is a static enum, why not just add it as an enum_type instead of doing an API call?

Reply:
You mean configure the argument in _params.py with something like `get_enum_type(vm_size_types)`? Seems like that should be added too. I was just trying to make minimal changes since this completer is shared by three arguments across `az acs` and `az aks`.

How does that interact with `completer=get_vm_size_completion_list`? The behavior in this PR means that the list of VM sizes is dynamic, depending on a `--location` argument or the user's subscription locations.

Reply:
Ah, okay, that makes sense.